### DISCUSSION TITLE
字数统计不是很对劲

### DISCUSSION THREAD

**Discussion by 2BeKindPLZ** [sjtug/SJTUThesis#1069] [提问]
论文写了一小丢丢，也不知道写了多少个字，但是pdf晃眼一看百多个还是有的吧...想看看字数统计
然后发现统计出来只有59个字=_=
不知道是不是我哪里整的有问题呀？
![075df52b90cb44dd6850e8d7a65abbe](https://github.com/user-attachments/assets/1cc86bdb-2093-4206-90a1-5b2fcea4fdf1)
这里是编译log文件
[main.log](https://github.com/user-attachments/files/17912450/main.log)

**Comment by LogCreative** [comment]
可能你在使用 Windows XP，把下面这一行
https://github.com/sjtug/SJTUThesis/blob/8580fa40763e6cb209116bd5da9e5b53cf4f064f/Compile.bat#L80
改成
```bat
	findstr /R /C:"\\documentclass\[[^\[]*lang *= *en" %THESIS%.tex > nul
```
试一下？

**Reply by 2BeKindPLZ** [reply]
啊0.0
不好意思忘了在问题里说了，我用的是Win10的系统，然后是在命令提示符里把路径切换到compile.bat所在的文件夹，随后运行的Compile.bat wordcount得到的问题里所展现的情况
刚刚把compile.bat里的文件做了修改，但字数统计输出的结果还是一样的

**Reply by LogCreative** [reply]
这次运行还有没有 `FINDSTR: Cannot open \C:\\documentclass` 这个提示吗，想确认一下

那字数统计应该没什么问题，要注意只有正文部分是计入字数的，摘要等都不计入字数。
https://github.com/sjtug/SJTUThesis/blob/8580fa40763e6cb209116bd5da9e5b53cf4f064f/main.tex#L50-L61

**Reply by 2BeKindPLZ** [reply]
> 这次运行还有没有 `FINDSTR: Cannot open \C:\\documentclass` 这个提示吗，想确认一下
> 
> 那字数统计应该没什么问题，要注意只有正文部分是计入字数的，摘要等都不计入字数。
> 
> https://github.com/sjtug/SJTUThesis/blob/8580fa40763e6cb209116bd5da9e5b53cf4f064f/main.tex#L50-L61

FINDSTR: Cannot open \C:\\documentclass 这个提示是没有了的
我写的是正文里的章节，用\input指令插在main.tex中的。指令在\mainmatter后面，\printbibliography之前，没有被%TC:ignore 和 %TC:endignore包含进去，但字数没有被统计

**Reply by LogCreative** [reply]
你复制到 Word 里看一看字数统计？那可能是 `wordcount` 的算法问题？

**Reply by 2BeKindPLZ** [reply]
好嘞，好像暂时也只好这么操作了，谢谢哈！

**Reply by LogCreative** [reply]
理论上 Word 里的统计结果与 wordcount 应该差距不大的，SJTUThesis 只是调用了 `wordcount` 这个工具做字数统计，工具本身我们这边是没有涉及的，你可以多试试哈

另外那个 `FINDSTR: Cannot open \C:\\documentclass` 提示准备在 #1070 中解决。
